### ISSUE TITLE
[sklearn#34649] Add option to drop infrequent categories in OneHotEncoder

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#301]
**Mirrored from [scikit-learn#34649](https://github.com/scikit-learn/scikit-learn/issues/34649)**

Original title: Add option to drop infrequent categories in OneHotEncoder
Original labels: New Feature, Needs Triage
Created: 2026-08-04T18:02:54Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

Currently, `OneHotEncoder(min_frequency=5)` groups all categorical levels with a cardinality (count) of less than 5 into a single new category "infrequent".
I would like to have an option, e.g. `drop_infrequent : bool` with which the infrequent categories are actually dropped: the column for the infrequent group is not added in the returned `transform(X)`.

### Describe your proposed solution

```py
```py
import numpy as np
import polars as pl
from sklearn.preprocessing import OneHotEncoder

X_train = pl.DataFrame({"a": ["a"]*9 + ["b"] + ["c"]*10, "b": ["a"]*5 + ["b"]*5 + ["c"]*5 + ["d"]*5})
X_test = pl.DataFrame({"a": ["a"] + ["b"] + ["c"]*2, "b": ["a"]+["b"]+["c"]+["d"]})

m1 = OneHotEncoder(min_frequency=2, sparse_output=False).fit(X_train)
m1.transform(X_test)
# array([[1., 0., 0.,   1., 0., 0., 0.],  <-- 3. column is infrequent
#        [0., 0., 1.,   0., 1., 0., 0.],
#        [0., 1., 0.,   0., 0., 1., 0.],
#        [0., 1., 0.,   0., 0., 0., 1.]])

m1 = OneHotEncoder(drop_infrequent=True, min_frequency=2, sparse_output=False).fit(X_train)
m1.transform(X_test)
# array([[1., 0.,   1., 0., 0., 0.],  
#        [0., 0.,   0., 1., 0., 0.],
#        [0., 1.,   0., 0., 1., 0.],
#        [0., 1.,   0., 0., 0., 1.]])
```

### Additional context

See https://github.com/scikit-learn/scikit-learn/pull/16018#issuecomment-1058435825.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/122 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
